Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenu.h

```diff
@@ -44,8 +44,12 @@ namespace AzToolsFramework
             MenuItem(QMenu* menu);
 
             MenuItemType m_type;
-            QAction* m_action;
-            QMenu* m_submenu;
+
+            union
+            {
+                QAction* m_action;
+                QMenu* m_submenu;
+            };
```

Comment:
suggestion/update: It might be better these days to use an `AZStd::variant`, much safer and easier not to shoot yourself in the foot with 😅

Reply:
Makes sense, I've replaced it.